spec/controllers/api/v1/trucking/driver_commissions_history_controller_spec.rb

```diff
@@ -0,0 +1,148 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::Trucking::DriverCommissionsHistoryController,
+               :type => :controller do
+  let(:driver_commission_history) do
+    [
+      {
+        :driver_id => Faker::Number.number(4),
+        :backhauls => 'BACKHAULS',
+        :freight_revenue => Faker::Number.number(4),
+        :customer_id => 0001,
+        :delivery_date => Date.new(2012, 01, 01),
+        :driver_rate => Faker::Number.number(4),
+        :freight_bill_number => Faker::Number.number(4),
+        :fuel_revenue => Faker::Number.number(4),
+        :fuel_surcharge => Faker::Number.number(4),
+        :layover => 'layover',
+        :movement_type => 'MOVE TYPE',
+        :other_hour_dollars => Faker::Number.number(4),
+        :order_number_text => Faker::Number.number(4),
+        :origin_name => 'ORIGIN NAME',
+        :paid_date => Date.new(2012, 01, 01),
+        :revenue => Faker::Number.number(4),
+        :split_rate => Faker::Number.number(4),
+        :total_freight_revenue => Faker::Number.number(4)
+      }
+    ]
+  end
+  before do
+    allow(DriverCommissionHistoryAdapter)
+      .to receive(:all) { driver_commission_history }
+    allow(DriverCommissionHistoryAdapter)
+      .to receive(:by_paid_date) { driver_commission_history }
+  end
+
+  describe 'GET index' do
+    before { index }
+    subject(:index) { get :index, :format => format }
+    context 'when format is json' do
+      let(:format) { :json }
+      it 'responds with 200' do
+        expect(response.status).to eq(200)
+      end
+      it 'responds with json' do
+        expect(response.content_type).to eq('application/json')
+      end
+      it 'body is not a null string' do
+        expect(response.body).not_to eq('null')
+      end
+      it 'body is parseable JSON' do
+        expect { JSON.parse(response.body) }.not_to raise_error
+      end
+      describe 'parsed JSON schema' do
+        let(:json) { JSON.parse(response.body, :symbolize_names => true) }
+        it 'is an array' do
+          expect(json).to be_kind_of(Array)
+        end
+        describe 'first element' do
+          subject(:first_element) { json.first }
+
+          it { expect(first_element).to be_kind_of(Hash) }
+          it { expect(first_element).to have_key(:driver_id) }
```

Comment:
Could you write this as `{ is_expected.to have_key(:driver_id) }`? This would save the repeated expect(first_element).